src/address_allocator.rs

```diff
@@ -63,13 +66,30 @@ impl AddressAllocator {
         policy: AllocPolicy,
     ) -> Result<RangeInclusive> {
         let constraint = Constraint::new(size, alignment, policy)?;
-        self.interval_tree.allocate(constraint)
+        let allocated = self.interval_tree.allocate(constraint)?;
+        self.used = self
+            .used
+            .checked_add(allocated.len() as usize)
+            .expect("Failed to calculate used memory");
```

Comment:
I am not sure that expect is the best solution here. I think it would be better to use `ok_or(Error::Overflow)` and propagate the error with ?